nvme-topology.c

```diff
@@ -686,10 +686,16 @@ int nvme_logical_block_size_from_ns_char(const char *dev)
 		return -EINVAL;
 
 	s = nvme_get_ctrl_attr(path, "logical_block_size");
-	if (!s)
+	if (!s) {
+		free(path);
```

Comment:
The free could be placed before the 'if, which avoids to add the free twice.

Reply:
You're right, I updated the PR